app/experimenter/experiments/tests/test_normandy.py

```diff
@@ -0,0 +1,45 @@
+import mock
+from requests.exceptions import RequestException, HTTPError
+from django.test import TestCase
+from experimenter.experiments.normandy import (
+    NormandyError,
+    make_normandy_call,
+    get_recipe,
+)
+from experimenter.experiments.tests.mixins import MockNormandyMixin
+
+
+class TestMakeNormandyCall(MockNormandyMixin, TestCase):
+
+    def test_sucessful_call(self):
+        mock_response_data = {"detail": "Not found."}
+        mock_response = mock.Mock()
+        mock_response.json = mock.Mock()
+        mock_response.json.return_value = mock_response_data
+        mock_response.raise_for_status = mock.Mock()
+        mock_response.raise_for_status.side_effect = None
+        self.mock_normandy_requests_get.return_value = mock_response
+
+        response_data = make_normandy_call("/url/")
+
+        self.assertEqual(response_data, mock_response_data)
+
+    def test_make_normandy_call_with_request_exception(self):
+        self.mock_normandy_requests_get.side_effect = RequestException()
+
+        with self.assertRaises(NormandyError):
+            make_normandy_call("/url/")
```

Comment:
yup, will do!

Reply:
@mythmon Not completely done with PR changes... but realized that Normandy Errors aren't raised with strings like "Error calling Normandy API" and they're only logged, so I modified tests to check for those log outputs. It isn't exactly what you asked, but was wondering if it's okay..

Reply:
The test you wrote will work, and it makes sure the right error case is being handled. On the other hand, it is kind of awkward. The error thrown by a method should tell you what broke, instead of having to rely on logging. Many times you'll only have access to the errors and not the logs (such as in Sentry).

Because of this, I suggest changing the way you handle errors to make them easier to distinguish by the code catching the exception. One way you could do that would be to add the error strings to the exceptions you're throwing. Another way would be to use entirely separate classes for each type of error. This makes it much clearer what is failing. You can see [an example of this pattern in Normandy](https://github.com/mozilla/normandy/blob/028960f99367e069e6bc4ca53fcc13c6393f19d2/normandy/recipes/signing.py#L157-L170).